### PR TITLE
Optional random seed as user input

### PR DESCRIPTION
**Pull request by HenningSE**
With this PR we will give users the option to set the random seed manually. If you want to use this function, you need to set `deterministic_seed` to `False` and `user_defined_random_seed` to the seed you want to use.

### CODE REVIEW

**Review by dachengx — CHANGES_REQUESTED** [review]
Thanks @HenningSE . Please add a test for this feature.

**Comment by HenningSE** [comment]
Thanks for the suggestion @dachengx. I added the following tests:

- test if all fuse plugins get the user defined seed
- test if all fuse plugins with a rng have a seed value defined
- test if we can intercept negative random seeds
- test if the run number changes the deterministic random seed
- test if a changed config arg changes the deterministic random seed

The last two tests are not specific to the changes in this PR but I think it does not hurt us to have them.

File: tests/test_plugin_random_seed.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            {
+                "deterministic_seed": False,
+                "user_defined_random_seed": -42,
+            }
+        )
+
+        # Lets check the random seed for all fuse plugins
+        for key in self.all_registered_fuse_plugins.keys():
+
+            try:
```
Comment (dachengx) [inline tests/test_plugin_random_seed.py:97]: Please use `with self.assertRaises(AssertionError):` but not barely `try except`.

**Review by dachengx — CHANGES_REQUESTED** [review]
Thanks @HenningSE . All good with a small potential improvement.

File: tests/test_plugin_random_seed.py (hunk 1 above)
Comment (HenningSE) [reply tests/test_plugin_random_seed.py]: Ahh nice! I did not know this feature yet! For plugins that have no seed I need to raise the AssertionError manually so the test will not fail.

**Review by dachengx — APPROVED** [review]

**Pull request merged**